src/async/smtp/MCSMTPSendWithDataOperation.cpp

```diff
@@ -84,3 +84,9 @@ void SMTPSendWithDataOperation::main()
     }
     setError(error);
 }
+
+void SMTPSendWithDataOperation::cancel()
+{
+    SMTPOperation::cancel();
+    session()->session()->cancelMessageSending();
```

Comment:
Maybe that's a good way to check if the cancel happened during connection after all ...